crates/client/rpc/src/lib.rs

```diff
@@ -1237,7 +1211,25 @@ where
     /// the state of the network as a result of the block's execution. This can include a confirmed
     /// state update or a pending state update. If the block is not found, returns a
     /// `StarknetRpcApiError` with `BlockNotFound`.
-    fn get_state_update(&self, block_id: BlockId) -> RpcResult<StateUpdate> {
+    fn get_state_update(&self, block_id: BlockId) -> RpcResult<MaybePendingStateUpdate> {
+        if is_pending_block(block_id) {
+            let state_diff = StateDiff {
+                storage_diffs: Vec::new(),
+                deprecated_declared_classes: Vec::new(),
+                declared_classes: Vec::new(),
+                deployed_contracts: Vec::new(),
+                replaced_classes: Vec::new(),
+                nonces: Vec::new(),
+            };
```

Comment:
So we return empty state diff in case of pending state?
I don't think this is the expected behavior. We should return the state changes generated by this set of pending txs

Reply:
@apoorvsadana @antiyro to be honest I don't even think this one is doable this way.
We could pre-execute the ready tx and gather the storage changes, but we can't guarantee they will be executed in the actual order the sequencer will chose later when producing the block

Reply:
So do we have any consensus how to get it properly?

Regarding `pre-execution` is there any way that I can get state diff from all the pending txs?

Reply:
> Regarding pre-execution is there any way that I can get state diff from all the pending txs?

Yep, by executing them with the runtime API, we will be able to reuse this logic: https://github.com/keep-starknet-strange/madara/pull/1411